crates/bevy_ecs/src/entity/clone_entities.rs

```diff
@@ -147,10 +147,10 @@ impl<'a, 'b> ComponentCloneCtx<'a, 'b> {
         if self.target_component_written {
             panic!("Trying to write component '{short_name}' multiple times")
         }
-        if !self
+        if self
             .component_info
             .type_id()
-            .is_some_and(|id| id == TypeId::of::<T>())
+            .is_none_or(|id| id != TypeId::of::<T>())
```

Comment:
Yeah I saw this come up a few times but was confused why it would fail CI locally for me and then pass on GitHub. Glad to see it resolved.